docs/tests.md

```diff
@@ -0,0 +1,454 @@
+
+This section provides an overview of the testing architecture for our Python
+library, which employs two main testing approaches:
+
+- **Known Answer Tests (KATs):** These tests, standard in cryptography, verify
+  the correctness of our implementation by comparing outputs with known results
+  for specific inputs. This approach ensures that cryptographic primitives and
+  algorithms are implemented correctly.
+- **Doctests:** These tests live within the code's docstrings, serving a dual
+  purpose. They illustrate how to use functions and classes through practical
+  examples, while simultaneously ensuring the code behaves as documented.
+
+Both KATs and Doctests are powered by the `pytest` framework. Doctests,
+specifically, leverage Python's built-in `doctest` module.
+
+## KAT Tests: Understanding the Process
+
+A Known Answer Test (KAT) in cryptography involves comparing the output of an
+algorithm implementation against a set of publicly available parameters and
+expected values. These expected values serve as reference points for
+correctness. Our KAT testing framework uses this approach to ensure that any
+modifications to the library do not disrupt existing functionality.
+
+Importantly, while pre-calculated parameters and values are common, some
+researchers provide source code for their implementations, serving as **KAT
+generators**. These generators allow for dynamic KAT value generation.
+
+Unlike isolated doctests, KATs take a broader approach by comparing outputs
+across entire estimators, often involving complex and computationally intensive
+operations. To optimize this process, we serialize expected outputs for specific
+predefined inputs. These serialized results streamline subsequent test runs,
+enabling efficient comparisons and saving valuable development time.
+
+Here's a breakdown of how our KAT testing process works:
+
+```mermaid
+flowchart LR;
+  katgen("`KAT Generators`")
+  inputs("`Inputs`")
+  preeo("`Precompute expected outputs`");
+  ser("`Serialize and store inputs/expected outputs`")
+  estimator("`Library estimation`")
+  oc("`TEST:
+    Outputs comparison`")
+
+  katgen --> preeo;
+  inputs --> preeo;
+  preeo --> ser;
+  ser -- inputs --> estimator;
+  estimator -- actual output --> oc;
+  ser -- expected output --> oc;
+```
+
+1. **KAT Generator Functions:** We utilize KAT generator functions located in
+   the `tests/external_estimators` directory. These functions have hardcoded
+   inputs as they are not intended for regular modification. Both input
+   parameters and their corresponding outputs (referred to as `expected_outputs`
+   as they act as reference values) are serialized into the `tests/kat.yaml`
+   file for later use during estimator testing.
+
+2. **Internal Estimation Functions:** With a collection of inputs and their
+   expected outputs, we define how these inputs should be processed within our
+   library. This is achieved through internal estimation functions found in the
```

Comment:
"we define how these inputs should be processed within our  library" --> we define how these inputs should be processed within our library in order to match the configuration of the external estimator. 

Does it make sense @Diego?

Reply:
Yes, I agree with the change.